### PR TITLE
Fix issue #450, SVGs using a style tag, has styles ignored

### PR DESCRIPTION
**Pull request by antman3351**
Let's try again....

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ mPDF 8.0.x
 * Compound classes selector support, like `.one.two` or `div.message.special` (#538, @peterdevpl)
 * Fixed CMYK colors in text-shadow (#1115, @lexilya)
 * Skip non supported wrappers when resolving paths (#1204, @MarkVaughn)
-
+* Fixed SVGs using a style tag, has styles ignored ( Requires ext-dom ) (#450, @antman3351)
```
Comment (finwe) [inline CHANGELOG.md:25]: Please keep the empty line

File: src/Image/Svg.php [hunk 2]
```diff
@@ -2824,6 +2824,93 @@ function svgWriteString($content)
...
+						if ($node->hasChildNodes()) {
+							$setStylesInline($node);
+						}
+					}
+				}
+			};
+
+			$setStylesInline($xml);
+
+		} catch (\Exception $e) {
```
Comment (finwe) [inline src/Image/Svg.php:2907]: Where exactly can the exception be thrown? Can't you surround only relevant code portions to reduce indentation?

Also, catching \Exception directly is not recommended, DOM methods throw an \DOMException, so you should catch that only to prevent catching unrelated exceptions, especially when you throw the exception details away in the catch block.

File: tests/Issues/issue450Test.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+class Issue450Test extends \Mpdf\BaseMpdfTest
+{
+
+	/**
+	 * SVGs with global CSS styles, had the styles ignored
+	 */
+	public function testSvgWithGlobalStyles()
+	{
+		// Copied from  Mpdf\Image\MetricsGeneratorTest
```
Comment (finwe) [inline tests/Issues/issue450Test.php:27]: How is this comment relevant?

File: tests/Issues/issue450Test.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		$otl = Mockery::mock(Otl::class);
+		$cssManager = Mockery::mock(CssManager::class);
+		$imageProcessor = Mockery::mock(ImageProcessor::class);
+		$this->sizeConverter = $this->createMock(SizeConverter::class); // Mockery::mock(ColorConverter::class); <-- doesn't work for me !??!
```
Comment (finwe) [inline tests/Issues/issue450Test.php:31]: How are these comments relevant? Either find out why Mockery doesn't work and fix it, or do not comment at all.

Comment (finwe) [reply tests/Issues/issue450Test.php:31]: Or state the reason of not working.

**Comment by antman3351** [comment]
> Comments to the code are in the review. Also please fix the failing CI.

Fail CI is because of old (unsupported) version of PHPUnit isn't compatible with PHP 7.4. ( when calling $this->createMock( ... ); )

File: src/Image/Svg.php (hunk 2 above)
Comment (antman3351) [reply src/Image/Svg.php:2907]: I thought DOMDocument was throwing an Exception, in fact it's issuing a Error (for malformed XML from another test) that PHPUnit was converting to an Exception.
I'll silence the Error as it's not much help in a PDF generation. `$xml->loadXML($data, LIBXML_NOERROR )`, the other option is to register an error handler to convert the Error to Exception. 

Surrounding the whole block I don't think is too bad as it's an "atomic" operation any Exception means something has gone wrong during the transformation. I prefer a SVG with missing colours to no SVG.  (Anyway I'll remove the try & catch)

I agree the indentation is terrible, I'll invert the ifs in the loop.

File: tests/Issues/issue450Test.php (hunk 3 above)
Comment (antman3351) [reply tests/Issues/issue450Test.php:27]: As I'm not familiar with the mpdf code base, if there's something wrong at least you know where I got it from 🤨

**Comment by finwe** [comment]
Thanks for a push, merged in 9f67876
